api_tests/nodes/views/test_node_detail.py

```diff
@@ -496,7 +496,7 @@ def test_node_shows_related_count_for_linked_by_relationships(self, app, user, p
         res = app.get(url)
         assert 'count' in res.json['data']['relationships']['linked_by_nodes']['links']['related']['meta']
         assert 'count' in res.json['data']['relationships']['linked_by_registrations']['links']['related']['meta']
-        assert res.json['data']['relationships']['linked_by_nodes']['links']['related']['meta']['count'] == 0
```

Comment:
I think this test was actually incorrect and upgrading Django fixed it, there should be a `linked_node` there.

Reply:
hmm, `RegistrationSerializer` inherits from `NodeSerializer`, and both defines their own `linked_by_nodes`, thus I don't think this is a DRF MRO issue since child/parent override is fine. [Here](https://github.com/encode/django-rest-framework/pull/6980#issuecomment-541236421) is the example of the MRO issue where sibling overriding order was the problem.

As for this test, my understanding is that `res.json['data']['relationships']['linked_by_nodes']['links']['related']['meta']['count'] == 0` should be true since the the private node has been deleted in the previous round. Withdrawing the registration shouldn't affect this count.

Reply:
Update: finally figured this out. This is due to `registration.registered_from` still keeps the old object. Somehow, after withdrawn, the old object overrides the new one. The fix is to reload the `registration`.

```python
>>>>>>>>> After private node deletion
public = (1845, 'gvckz', False, None)
private = (1846, '9mfd5', True, datetime.datetime(2022, 10, 5, 3, 7, 51, 792315, tzinfo=<UTC>))
registered_from = (1846, '9mfd5', False, None)

>>>>>>>>> After registration withdrawn
public = (1845, 'gvckz', False, None)
private = (1846, '9mfd5', False, None)
```

Fyi, the query from `get_linked_by_nodes_count` looks good

```sql
INFO: calling get_linked_by_nodes_count in NodeDetailSerializer
[api.nodes.serializers]  WARNING: query = SELECT "osf_noderelation"."id", "osf_noderelation"."created", "osf_noderelation"."modified", "osf_noderelation"."_id", "osf_noderelation"."parent_id", "osf_noderelation"."child_id", "osf_noderelation"."is_node_link", "osf_noderelation"."_order" FROM "osf_noderelation" INNER JOIN "osf_abstractnode" T3 ON ("osf_noderelation"."parent_id" = T3."id") WHERE ("osf_noderelation"."child_id" = 1845 AND "osf_noderelation"."is_node_link" AND NOT T3."is_deleted" AND T3."type" = osf.node) ORDER BY "osf_noderelation"."_order" ASC
```